content-src/components/StartupOverlay/StartupOverlay.jsx

```diff
@@ -15,12 +15,24 @@ export class _StartupOverlay extends React.PureComponent {
 
     this.state = {
       emailInput: "",
-      overlayRemoved: false
+      overlayRemoved: false,
+      flowId: "",
+      flowBeginTime: ""
     };
     this.initScene();
   }
 
-  initScene() {
+  async initScene() {
+    try {
+      let response = await fetch(`${this.props.fxa_endpoint}/metrics-flow`);
```

Comment:
Do this fetch stuff in componentWillUpdate, that's when the props from the connected Prefs will be ready. Any earlier than that and our store isn't ready.
Also add a flag to make sure we're not doing multiple fetchs when we update:

```js
  async componentWillUpdate() {
    try {
      if (!this.props.fxa_endpoint || this.didFetch) {
        return;
      }
      this.didFetch = true;
      const response = await fetch(`${this.props.fxa_endpoint}/metrics-flow`);
      if (response.status === 200) {
        const {flowId, flowBeginTime} = await response.json();
        this.setState({flowId, flowBeginTime});
      }
    } catch (error) {
      this.props.dispatch(ac.OnlyToMain({type: at.TELEMETRY_UNDESIRED_EVENT, data: {value: "FXA_METRICS_ERROR"}}));
    }
  }
```